text-generation-inference/tgi_env.py

```diff
@@ -0,0 +1,236 @@
+        os.environ["MAX_BATCH_SIZE"] = str(args.max_batch_size)
+        os.environ["HF_BATCH_SIZE"] = str(args.max_batch_size)
+
+    if args.revision:
+        os.environ["REVISION"] = str(args.revision)
+
+    return args
+
+
+def neuron_config_to_env(neuron_config):
+    with open(os.environ["ENV_FILEPATH"], "w") as f:
+        for env_var, config_key in env_config_peering:
+            f.write("export {}={}\n".format(env_var, neuron_config[config_key]))
+        max_input_length = os.getenv("MAX_INPUT_LENGTH")
+        if not max_input_length:
+            max_input_length = int(neuron_config["sequence_length"]) // 2
+            if max_input_length == 0:
+                raise Exception("Model sequence length should be greater than 1")
+        f.write("export MAX_INPUT_LENGTH={}\n".format(max_input_length))
+
+        # max_prefill_tokens = os.getenv("MAX_BATCH_PREFILL_TOKENS")
+        # if not max_prefill_tokens:
+        #     max_prefill_tokens = max_input_length
+        # f.write("export MAX_BATCH_PREFILL_TOKENS={}\n".format(max_prefill_tokens))
+        #
+        # max_batch_total_tokens = os.getenv("MAX_BATCH_TOTAL_TOKENS")
+        # if not max_batch_total_tokens:
+        #     max_batch_total_tokens = str(neuron_config["sequence_length"] * neuron_config["batch_size"])
+        # f.write("export MAX_BATCH_TOTAL_TOKENS={}\n".format(max_batch_total_tokens))
+
+
+def lookup_compatible_cached_model(model_id: str, revision: Optional[str]) -> Optional[Dict[str, Any]]:
+    # Reuse the same mechanic as the one in use to configure the tgi server part
+    # The only difference here is that we stay as flexible as possible on the compatibility part
+    entries = get_hub_cached_entries(model_id, "inference")
+
+    logger.debug("Found %d cached entries for model %s, revision %s", len(entries), model_id, revision)
+
+    for entry in entries:
+        if check_env_and_neuron_config_compatibility(revision, entry):
+            break
+    else:
+        entry = None
+
+    if not entry:
+        logger.debug(
+            "No compatible cached entry found for model %s, env %s, available cores %s, " "neuronxcc version %s",
+            model_id,
+            get_env_dict(),
+            available_cores,
+            neuronxcc_version,
+        )
+    else:
+        logger.info("Compatible neuron cached model found %s", entry)
+    return entry
+
+
+def check_env_and_neuron_config_compatibility(revision: Optional[str], neuron_config: Dict[str, Any]) -> bool:
+
+    logger.debug(
+        "Checking the provided neuron config %s is compatible with the local setup and " "provided environment",
+        neuron_config,
+    )
+
+    if revision:
+        checkpoint_revision = neuron_config["checkpoint_revision"]
+        if revision != checkpoint_revision:
+            logger.debug(
+                "Specified revision does not match with the model one (%s != %s)", revision, checkpoint_revision
+            )
+            return False
```

Comment:
You can remove this check: the revision in the neuron config is the one from the original model, that will be used to fetch the weights. It is not the neuron model revision passed as parameter to the script.